File: app/src/main/java/org/dhis2/utils/ObjectStyleUtils.java
package org.dhis2.utils;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.Color;
import android.graphics.drawable.Drawable;

import androidx.annotation.ColorRes;
import androidx.appcompat.content.res.AppCompatResources;
import androidx.core.content.ContextCompat;

import static android.text.TextUtils.isEmpty;

public class ObjectStyleUtils {

    public static Drawable getIconResource(Context context, String resourceName, int defaultResource) {
        if (defaultResource == -1) {
            return null;
        }
        Drawable defaultDrawable = AppCompatResources.getDrawable(context, defaultResource);

        if (!isEmpty(resourceName)) {
            Resources resources = context.getResources();
            String iconName = resourceName.startsWith("ic_") ? resourceName : "ic_" + resourceName;
            int iconResource = resources.getIdentifier(iconName, "drawable", context.getPackageName());

            Drawable drawable = AppCompatResources.getDrawable(context, iconResource);

            if (drawable != null)
                drawable.mutate();

            return drawable != null ? drawable : defaultDrawable;
        } else
            return defaultDrawable;
    }

    public static int getColorResource(Context context, String styleColor, @ColorRes int defaultColorResource) {
        if (styleColor == null) {
            return ContextCompat.getColor(context, defaultColorResource);
        } else {
            String color = styleColor.startsWith("#") ? styleColor : "#" + styleColor;
            int colorRes;
            if (color.length() == 4)
                return ContextCompat.getColor(context, defaultColorResource);
            else
                colorRes = Color.parseColor(color);

            return colorRes;
        }
    }
}

File: app/src/main/java/org/dhis2/utils/ObjectStyleUtils.kt
package org.dhis2.utils

import android.content.Context
import android.content.res.Resources
import android.graphics.drawable.Drawable

import androidx.appcompat.content.res.AppCompatResources

object ObjectStyleUtils {

    fun getIconResource(context: Context, resourceName: String?, defaultResource: Int): Drawable? {
        val defaultDrawable = AppCompatResources.getDrawable(context, defaultResource)

        return if (resourceName != null) {
            val resources = context.resources
            val iconName = if (resourceName.startsWith("ic_")) resourceName else "ic_$resourceName"
            val iconResource = resources.getIdentifier(iconName, "drawable", context.packageName)

            val drawable = AppCompatResources.getDrawable(context, iconResource)

            drawable ?: defaultDrawable
        } else
            defaultDrawable
    }

}
